send/generic_send

```diff
@@ -83,12 +92,25 @@ if [ $? -ne 0 ]; then
 	exit 255
 fi
 
+TEMPORARY_FILE="/tmp/${SERVICE_NAME}-$$.tar"
 if [ -d "$SERVICE_FILES_FOR_DESTINATION" ]
 then
-	tar -c -C "$SERVICE_FILES_FOR_DESTINATION" . -C "$SERVICE_FILES_DIR"  --exclude="_destination" .  -C "$TMP_HOSTNAME_DIR" .  | timeout -k $TIMEOUT_KILL $TIMEOUT $TRANSPORT_COMMAND $HOST $SLAVE_COMMAND
+	tar -c -f "$TEMPORARY_FILE" -C "$SERVICE_FILES_FOR_DESTINATION" . -C "$SERVICE_FILES_DIR"  --exclude="_destination" .  -C "$TMP_HOSTNAME_DIR" . 
 else
-	tar -c -C "$SERVICE_FILES_DIR"  --exclude="_destination" .  -C "$TMP_HOSTNAME_DIR" .  | timeout -k $TIMEOUT_KILL $TIMEOUT $TRANSPORT_COMMAND $HOST $SLAVE_COMMAND
+	tar -c -f "$TEMPORARY_FILE" -C "$SERVICE_FILES_DIR"  --exclude="_destination" .  -C "$TMP_HOSTNAME_DIR" . 
 fi
+
+if [ $? -ne 0 ]; then
+	echo "Can't archive service files to $TEMPORARY_FILE" >&2
+	exit 255
+fi
+
+if [ "$DESTINATION_TYPE"="url" ]; then
+	timeout -k $TIMEOUT_KILL $TIMEOUT $TRANSPORT_COMMAND --data-binary @$TEMPORARY_FILE $HOST
```

Comment:
curl can read data from STDIN using following syntax. Therefore there is no need to create extra file (\$TEMPORARY_FILE)
```bash
curl -i -H Content-Type:application/tar -f -X PUT --data-binary -
```